pkg/requirements/requirements.go

```diff
@@ -48,3 +38,59 @@ func CurrentRequirements(tmpDir string) (string, error) {
 	}
 	return requirementsFile, nil
 }
+
+func ReadRequirements(path string) ([]string, error) {
+	fh, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	// Use scanner to handle CRLF endings
+	scanner := bufio.NewScanner(fh)
+	scanner.Split(scanLinesWithContinuations)
+	requirements := []string{}
+	for scanner.Scan() {
+		requirementsText := strings.TrimSpace(scanner.Text())
+		if len(requirementsText) == 0 {
+			continue
+		}
+		requirements = append(requirements, scanner.Text())
+	}
+	return requirements, nil
+}
+
+func scanLinesWithContinuations(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	advance = 0
+	token = nil
+	err = nil
+	inHash := false
+	for {
+		if atEOF || len(data) == 0 {
+			break
+		}
+		if token == nil {
+			token = []byte{}
+		}
+		if data[advance] == '#' {
+			inHash = true
+		}
+		if data[advance] == '\n' {
```

Comment:
I think based on the comment on line 47, this should handle CRLF characters too.

Reply:
Resolved in [16a4ca8](https://github.com/replicate/cog/pull/2230/commits/16a4ca850e8dd3b46527aadeec01c6b3498cbee6)